scripts/import-cp-contexts.sh

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+arg_names=""
+arg_kubeconfig=""
+arg_context=""
+arg_ip_addr=""
+arg_merge=false
+arg_output=""
+arg_verbose=true
+
+
+display_help() {
+  cat << EOF
+Usage: $0 [--kubeconfig <filename>] [--context <name>] [--names <list of names>] [--replace-localhost <address>] [--merge] [-o <filename>] [-V] [-X]
+--kubeconfig <filename>       use the specified kubeconfig
+--context <name>              use the specified context
+--names <name1>,<name2>       comma separated list of KubeFlex Control Planes names to import, instead of default *all*
+--replace-localhost <address> replaces server addresses "127.0.0.1" with <address>
```

Comment:
Does the replacement have to be an address (can not be a domain name)?

Reply:
No it can be a domani but I have not found an example

Reply:
I think that it is important to not obscure the possibility that the replacement can be a domain name. What sort of example are you looking for? It is easy to make up domain names. We might use "core.example.com", for example.

Reply:
My question is not about "127.0.0.1", you explained that previously. My question is about the string that "127.0.0.1" is being replaced with. The thing that is described here as `<address>`.

Why does any replacement at all work? Won't the client get an X.509 certificate for a Subject whose Common Name is "127.0.01" regardless of kubeconfig editing?

If replacement with an IP address works, why does replacement with a domain name that resolves to that IP address not work?

Reply:
The reason why it works is because for kind and k3s we use SSL passthrough.
A domain will also work for the passthrough.
However, there is a specific problem that this approach is trying to solve.
a WEC kind/k3s cluster on a machine B will not be able to access the ITS in the the kind/k3s cluster of machine A unless:
1. the ITS is of type host
2. the ITS kubeconfig contains the IP address (not the domain name) of the machine A
AFSK this is the only scenario where the IP swap is needed

Reply:
the domain replacement will technically work but the WEC cluster in machine B does not have a way to route to the domain, the machine B can route by adding the domain/ip in it /etc/hosts but the kind/k3s cluster running on machine B does not use the /etc/hosts... so only an ip actually is known to work to access the its1. BTW, id did mentioned that one could use a domain in help print out.

Reply:
Of course a domain name is only useful if the clients using it can resolve it to the intended IP address. That is totally ordinary, not a failing of our system.

Putting domain names in `/etc/hosts` is a hack with limited and system-dependent efficacy.

I expect that an organization that would use KubeStellar for real might have "real" domain names (ones that any non-air-gapped client can resolve) or the ability to create them if needed.

So it looks to me like, as far as KubeStellar is concerned, the replacement _can_ be a domain name; it is naturally the user's responsibility to use a domain name that works, if they choose to use a domain name.

So I think that the documentation, command line flag help, and code should be worded to cover the case of domain name as well as IP address. I think that the word "host" might be usable to cover both cases.

Reply:
I replaced with host